internal/validation/validation.go

```diff
@@ -1,38 +1,17 @@
 package validation
 
 import (
-	"fmt"
-	"net/url"
-)
-
-// IsURLWithNoFragment is a SchemaValidateFunc which tests if the provided value
-// is of type string and a valid URL with no fragment.
-func IsURLWithNoFragment(i interface{}, k string) (warnings []string, errors []error) {
-	v, ok := i.(string)
-	if !ok {
-		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
-		return
-	}
-
-	if v == "" {
-		errors = append(errors, fmt.Errorf("expected %q url to not be empty, got %v", k, i))
-		return
-	}
+	"strings"
 
-	u, err := url.Parse(v)
-	if err != nil {
-		errors = append(errors, fmt.Errorf("expected %q to be a valid url, got %v: %+v", k, v, err))
-		return
-	}
-
-	if u.Host == "" {
-		errors = append(errors, fmt.Errorf("expected %q to have a host, got %v", k, v))
-		return
-	}
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
 
-	if u.Fragment != "" {
-		errors = append(errors, fmt.Errorf("expected %q to have a url with an empty fragment. %s", k, v))
+func IsURLWithHTTPSorEmptyString(i interface{}, s string) ([]string, []error) {
```

Comment:
Should this be `IsURLWithHTTPSOrEmptyString`?

Reply:
IMO no, as the "O" is not part of "HTTP" and it also reads easier.

A similar example was set in https://github.com/hashicorp/terraform-plugin-sdk/blob/8ffdd1ea219bc0bbb64c0e7a87bf081ad40ca3ee/helper/validation/web.go#L17.